lgc/builder/DescBuilder.cpp

```diff
@@ -417,8 +416,19 @@ Value *DescBuilder::CreatePtrDiff(llvm::Value *lhs, llvm::Value *rhs, const llvm
       !rhsType->isPointerTy() || rhsType->getPointerAddressSpace() != ADDR_SPACE_BUFFER_FAT_POINTER)
     return IRBuilderBase::CreatePtrDiff(lhs, rhs, instName);
 
-  std::string callName = lgcName::LateBufferPtrDiff;
-  return CreateNamedCall(lgcName::LateBufferPtrDiff, getInt64Ty(), {lhs, rhs}, Attribute::ReadNone);
+  // Create vararg function, so it can accept any argument types
```

Comment:
For consistency, can we handle this in the same way that other occurrences of the same problem are handled? That is, add mangling to the name by calling addTypeMangling().

Using varargs is a good idea, but I think we should do that everywhere, in a separate change. We could do this:
* ensure all uses of CreateNamedCall for intrinsics get changed to CreateIntrinsic
* come up with some special handling for NPI instrinsics
* change the implementation of CreateNamedCall to use varargs
* remove calls to addTypeMangling.

Reply:
Thanks, I missed that there is `addTypeMangling` for this purpose. It’s much less code now.